src/AppInstallerCLICore/Workflows/RepairFlow.cpp

```diff
@@ -340,11 +415,11 @@ namespace AppInstaller::CLI::Workflow
 
     void GetRepairInfo(Execution::Context& context)
     {
-        const auto& installer = context.Get<Execution::Data::Installer>();
-        InstallerTypeEnum installerTypeEnum = installer->EffectiveInstallerType();
+        InstallerTypeEnum installerTypeEnum = GetInstalledPackageInstallerType(context);

```

Comment:
Utilizing 'InstallerType' demands prior installer selection, leading to dependencies we aim to avoid for MSI/MSIX(non-store) repairs. An alternative could be splitting the method to separate the processes that require installer mapping from those that do not

Reply:
Same as previous comment, use InstallerType if exists, otherwise use installed type. Installed type is very limited. For example, if a nullsoft is installed, the detected installed type will be exe.